GitCommands/Git/ExecutableFactory.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using GitUIPluginInterfaces;
+using JetBrains.Annotations;
+
+namespace GitCommands
+{
+    public class ExecutableFactory
+    {
+        /// <summary>
+        /// Singleton accessor.
+        /// </summary>
+        public static ExecutableFactory Default { get; } = new ExecutableFactory();
+
+        public IExecutable Create([NotNull] Func<string> fileNameProvider,
+            [NotNull] string workingDir = "",
+            ExternalOperationExceptionFactory.Handling exceptionHandling = ExternalOperationExceptionFactory.Handling.OptionalBugReport,
+            [CanBeNull] string context = null)
+            => new Executable(fileNameProvider, workingDir, exceptionHandling, context);
+
+        public IExecutable Create([NotNull] string fileName,
+            [NotNull] string workingDir = "",
+            ExternalOperationExceptionFactory.Handling exceptionHandling = ExternalOperationExceptionFactory.Handling.OptionalBugReport,
+            [CanBeNull] string context = null)
+            => Create(() => fileName, workingDir, exceptionHandling, context);
```

Comment:
I only see one place where this is used. 
Could you explain why we need these two methods and not one?

`Func<string> fileNameProvider` 

This is dependency, but it's not clear why.